src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+
+    if smooth is True:
+        smooth_kwargs = {}
+        smooth_kwargs.setdefault("window_length", 55)
+        smooth_kwargs.setdefault("polyorder", 2)
+        x_data = np.linspace(pred_x_vals.min(), pred_x_vals.max(), 200)
+        x_data[0] = (x_data[0] + x_data[1]) / 2
+        hdi_interp = griddata(pred_x_vals, ci_data, x_data)
+        y_data = savgol_filter(hdi_interp, axis=0, **smooth_kwargs)
+    else:
+        idx = np.argsort(pred_x_vals)
+        x_data = pred_x_vals[idx]
+        y_data = ci_data[idx]
+
+    ci_lower = y_data[:, 0]
+    ci_upper = y_data[:, 1]
+
+    # upper and lower lines of credible interval
+    line_kwargs = copy(visuals.get("line", {}))
+    if line_kwargs is not False:
+        _, _, line_ignore = filter_aes(plot_collection, aes_by_visuals, "line", sample_dims)
+        plot_collection.map(
+            line_xy,
+            "lower_ci_line",
+            x=x_data,
+            y=ci_lower,
+            ignore_aes=line_ignore,
+            **line_kwargs,
+        )
+
+        plot_collection.map(
+            line_xy,
+            "upper_ci_line",
+            x=x_data,
+            y=ci_upper,
+            ignore_aes=line_ignore,
+            **line_kwargs,
+        )
+
+    # data prepration for `fill` visual
+    data = np.stack([x_data, ci_lower, ci_upper], axis=0)
+    da = xr.DataArray(
+        data,
+        dims=["kwarg", "index"],
+        coords={
+            "kwarg": ["x", "y_bottom", "y_top"],
+            "index": np.arange(len(x_data)),
+        },
+        name="ci_bounds",
+    )
+
+    # fill between lines of credible interval
+    fill_kwargs = copy(visuals.get("fill", {}))
+    if fill_kwargs is not False:
+        _, _, fill_ignore = filter_aes(plot_collection, aes_by_visuals, "fill", sample_dims)
+        plot_collection.map(
+            fill_between_y,
+            "fill",
+            data=da,
+            ignore_aes=fill_ignore,
+            color=colors[1],
+            alpha=0.4,
+            **fill_kwargs,
+        )
+
+    # mean line
+    mean_line_kwargs = copy(visuals.get("mean_line", {}))
+    if mean_line_kwargs is not False:
+        _, _, mean_line_ignore = filter_aes(
+            plot_collection, aes_by_visuals, "mean_line", sample_dims
+        )
+        plot_collection.map(
+            line_xy,
+            "mean_line",
+            x=pred_x_vals,
+            y=mean_data.y.values,
+            ignore_aes=mean_line_ignore,
+            **mean_line_kwargs,
+        )
+
+    # scatter plot
+    original_scatter_kwargs = copy(visuals.get("scatter", {}))
+    if original_scatter_kwargs is not False:
+        _, _, scatter_ignore = filter_aes(plot_collection, aes_by_visuals, "scatter", sample_dims)
+        plot_collection.map(
+            scatter_xy,
+            "scatter",
+            x=independent_data[independent_var].values,
+            y=target_data[target_var].values,
+            alpha=0.3,
+            color=contrast_gray_color,
+            width=0,
```

Comment:
all of these should be added through `setdefault` and only when on aesthetic is mapped to them. The scatter/line/area can be enough distinction between observed/mean/credible interval and we might want all to have a mapping from color to group like we did in https://arviz-plots.readthedocs.io/en/latest/tutorials/compose_own_plot.html#option-2-specific-functions-for-each-visual-element